webapp/channels/src/components/admin_console/database_settings.tsx

```diff
@@ -29,16 +51,16 @@ export default class DatabaseSettings extends AdminSettings {
         };
     }
 
-    getConfigFromState = (config) => {
+    getConfigFromState = (config: AdminConfig) => {
         // driverName and dataSource are read-only from the UI
 
         config.SqlSettings.MaxIdleConns = this.parseIntNonZero(this.state.maxIdleConns);
         config.SqlSettings.MaxOpenConns = this.parseIntNonZero(this.state.maxOpenConns);
         config.SqlSettings.Trace = this.state.trace;
         config.SqlSettings.DisableDatabaseSearch = this.state.disableDatabaseSearch;
         config.SqlSettings.QueryTimeout = this.parseIntNonZero(this.state.queryTimeout);
-        config.SqlSettings.ConnMaxLifetimeMilliseconds = this.parseIntNonNegative(this.state.connMaxLifetimeMilliseconds);
-        config.SqlSettings.ConnMaxIdleTimeMilliseconds = this.parseIntNonNegative(this.state.connMaxIdleTimeMilliseconds);
+        config.SqlSettings.ConnMaxLifetimeMilliseconds = this.parseIntNonZero(this.state.connMaxLifetimeMilliseconds);
+        config.SqlSettings.ConnMaxIdleTimeMilliseconds = this.parseIntNonZero(this.state.connMaxIdleTimeMilliseconds);
```

Comment:
I dont think we need to change the method here, lets keep the previous method but we can change the parseIntNonNegative param types